odoo/modules/module.py

```diff
@@ -465,22 +465,115 @@ def get_test_modules(module):
               if name.startswith('test_')]
     return result
 
-# Use a custom stream object to log the test executions.
-class TestStream(object):
-    def __init__(self, logger_name='odoo.tests'):
-        self.logger = logging.getLogger(logger_name)
-        self.r = re.compile(r'^-*$|^ *... *$|^ok$')
-    def flush(self):
-        pass
-    def write(self, s):
-        if self.r.match(s):
+
+class OdooTestResult(unittest.result.TestResult):
+    """
+    This class in inspired from TextTestResult (https://github.com/python/cpython/blob/master/Lib/unittest/runner.py)
+    Instead of using a stream, we are using the logger,
+    but replacing the "findCaller" in order to give the information we
+    have based on the test object that is running.
+    """
+
+    def log(self, level, msg, *args, test=None, exc_info=None, extra=None, stack_info=False, caller_infos=None):
+        """
+        :param test: the running test case
+        :param caller_infos: (fn, lno, func, sinfo) (logger.findCaller format)
+
+        other params are logger.log params
+
+        we dont use the logger.log method since it would be difficult to change findCaller
+        behaviour. This implementation uses the low_level makeRecord and handle methods
+        instead
+        """
+        logger = logging.getLogger((test or self).__module__)  # test should be always set
+        try:
+            caller_infos = caller_infos or logger.findCaller(stack_info)
+        except ValueError:
+            caller_infos = "(unknown file)", 0, "(unknown function)", None
+        (fn, lno, func, sinfo) = caller_infos
+        record = logger.makeRecord(logger.name, level, fn, lno, msg, args, exc_info, func, extra, sinfo)
+        logger.handle(record)
+
+    def getDescription(self, test):
+        """
+        Simplified version of TextTestResult getDescription
+        """
+        if isinstance(test, unittest.TestCase):
+            return "%s.%s" % (test.__class__.__qualname__, test._testMethodName)
+        return str(test)
```

Comment:
Is there any case where the test will not be a TestCase?

Also maybe explain why we're overriding this (I expect because we're using the module as logger so having that as part of `TestCase.__str__` is redundant for our runner / situation).

And only the `return str(test)` remains from TextTestResult's own getDescription, and it probably won't get invoked, so that seems like a debatable thing to docstringify.